gpm_api/tests/test_visualization/test_facetgrid.py

```diff
@@ -0,0 +1,229 @@
+        self,
+        orbit_dataarray_x4: xr.DataArray,
+    ) -> None:
+        """Test plotting orbit data without valid col or row argument"""
+
+        with pytest.raises(ValueError):
+            plot.plot_map(orbit_dataarray_x4, col=None)
+
+    def test_orbit_one_empty_subplot(
+        self,
+        orbit_dataarray: xr.DataArray,
+    ) -> None:
+        """Test plotting orbit data with one empty subplot"""
+
+        orbit_dataarray = expand_dims(orbit_dataarray, 3, channel=CHANNEL)
+        p = plot.plot_map(orbit_dataarray, col=CHANNEL, col_wrap=2)
+        save_and_check_figure(figure=p.fig, name=get_test_name())
+
+    def test_orbit_extent(
+        self,
+        orbit_dataarray_x4: xr.DataArray,
+    ) -> None:
+        """Test plotting orbit data while specifying extent"""
+
+        extent = [0, 20, 0, 20]
+        p = plot.plot_map(orbit_dataarray_x4, col=CHANNEL, col_wrap=2)
+        p.remove_title_dimension_prefix()
+        p.set_extent(extent)
+        save_and_check_figure(figure=p.fig, name=get_test_name())
+
+    def test_orbit_non_unique_index(
+        self,
+        orbit_dataarray_x4: xr.DataArray,
+    ) -> None:
+        """Test plotting orbit data with non-unique channel index"""
+
+        orbit_dataarray_x4[CHANNEL] = [0, 1, 1, 2]
+        with pytest.raises(ValueError):
+            plot.plot_map(orbit_dataarray_x4, col=CHANNEL, col_wrap=2)
+
+    def test_orbit_no_colorbar(
+        self,
+        orbit_dataarray_x4: xr.DataArray,
+    ) -> None:
+        """Test plotting orbit data without colorbar"""
+
+        p = plot.plot_map(orbit_dataarray_x4, col=CHANNEL, col_wrap=2, add_colorbar=False)
+        save_and_check_figure(figure=p.fig, name=get_test_name())
+
+    def test_orbit_horizontal_colorbar(
+        self,
+        orbit_dataarray_x4: xr.DataArray,
+    ) -> None:
+        """Test plotting orbit data with a horizontal colorbar"""
+
+        cbar_kwargs = {"orientation": "horizontal"}
+        p = plot.plot_map(orbit_dataarray_x4, col=CHANNEL, col_wrap=2, cbar_kwargs=cbar_kwargs)
+        save_and_check_figure(figure=p.fig, name=get_test_name())
+
+    def test_orbit_colorbar_ticks(
+        self,
+        orbit_dataarray_x4: xr.DataArray,
+    ) -> None:
+        """Test plotting orbit data with custom colorbar tick labels"""
+
+        cbar_kwargs = {"ticklabels": [42, 43, 44, 45]}
+        p = plot.plot_map(orbit_dataarray_x4, col=CHANNEL, col_wrap=2, cbar_kwargs=cbar_kwargs)
+        save_and_check_figure(figure=p.fig, name=get_test_name())
+
+    def test_orbit_extended_colorbar(
+        self,
+        orbit_dataarray_x4: xr.DataArray,
+    ) -> None:
+        """Test plotting orbit data with extended colorbar"""
+
+        cbar_kwargs = {"extend": "both", "extendfrac": 0.03}
+        p = plot.plot_map(orbit_dataarray_x4, col=CHANNEL, col_wrap=2, cbar_kwargs=cbar_kwargs)
+        save_and_check_figure(figure=p.fig, name=get_test_name())
+
+    def test_orbit_high_aspect_ratio(
```

Comment:
To remove. Cartopy does not allow aspect ratio different than 1. And optimize_layout destroy everything.
But `facet_aspect` works correctly for `plot_image` to derive `facet_width` based on the specified `faced_height` f I recall correctly ;)